### DISCUSSION TITLE
Add Pydal Snowflake support please

### DISCUSSION THREAD

**Discussion by KellerKev** [emmett-framework/emmett#435] [Ideas]
I recently add a snowflake adapter to pydal.

Could you help me adding snowflake support to Emmetts orm based on that please?

I would like to advertise Emmett within snowflake as a web framework.

Maybe you can give it a shot and I test it and see that I can kink out any details if it doesn't work. Then we could merge it and I can show my colleagues at snowflake how to use Emmett with their customers.

**Comment by gi0baro** [comment]
@Kkeller83 so, I think the time has finally came to review the pyDAL dependency/integration.

The main issues, as of today, with pyDAL are:
- it doesn't follow semantic versioning, so it would be impossible to pick a "stable" requirement in Emmett
- if I recall correctly, after 17.03 something changed with callbacks order in contrast with what Emmett needs
- the `orm` module currently applies several monkey-patches to pyDAL in order to make its features to be fully-working

Given the upper points, upgrading the dependency requires **a lot** a work. I actually have something in mind for #309, which in addition to the split of the actual `orm` module in a separate package, in my perspective should also produce a modular solution for adapters, so for instance there would no need to update the entire orm module just to add a specific engine (eg. having something like `emmett-orm` package and `emmett-orm-snowflake` independent).

So the question now is: how do we proceed until we reach that?
I think the easiest way is to open a PR also here, with the same code you put in pyDAL, as `emmett` relies on `pydal` structures, as you can see from the custom postgreSQL implementation here https://github.com/emmett-framework/emmett/blob/5cd622072dcebe7de22aacbe2eb8d4c60b75cf0f/emmett/orm/engines/postgres.py

If you manage to open a PR as a starting point I would be definitely available to check it and eventually finalise it for a merge and releasing it as "experimental".

**Comment by KellerKev** [comment]
Ok will give it a shot
